# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFontInfoLoader.cpp
# Commit: 0d746825f92a
# Full Hash: 0d746825f92a3039433aa275997c6e7f2ebfaf56
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1708768
# File Overlap Count: 2
# Description:
#   Bug 1708768 - Make InitFontList safe to call off-main-thread on macOS, and run font-list initialization on a separate thread during startup. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D115295
# ==============================================================================

diff -r fbe7310ee5f3 -r 0d746825f92a gfx/thebes/gfxFontInfoLoader.cpp
--- a/gfx/thebes/gfxFontInfoLoader.cpp	Tue May 18 11:10:43 2021 +0000
+++ b/gfx/thebes/gfxFontInfoLoader.cpp	Tue May 18 11:10:43 2021 +0000
@@ -161,10 +161,12 @@
     NS_ASSERTION(!gXPCOMThreadsShutDown,
                  "Bug 1508626 - Setting delay timer for font loader after "
                  "shutdown but before observer");
-    // Set up delay timer. We don't expect to be called with a delay once
-    // the timer already exists, but if that happens we'll just ignore the
-    // extra call and leave the existing timer to do its thing.
-    MOZ_ASSERT(!mTimer, "duplicate use of StartLoader() with delay?");
+    // Set up delay timer, or if there is already a timer in place, just
+    // leave it to do its thing. (This can happen if a StartLoader runnable
+    // was posted to the main thread from the InitFontList thread, but then
+    // before it had a chance to run and call StartLoader, the main thread
+    // re-initialized the list due to a platform notification and called
+    // StartLoader directly.)
     if (mTimer) {
       return;
     }